stargate-starter/src/test/java/io/stargate/starter/StarterTest.java

```diff
@@ -156,19 +140,74 @@ void testSeedsNotPresentThrows() {
 
   @Test
   void testDeveloperModeSetsDefaultSeedsAndSnitch() {
-    Starter starter = new Starter();
+    Starter starter = setupDefaultStarter();
     starter.developerMode = true;
-
-    starter.clusterName = "foo";
-    starter.version = "3.11";
-
-    assertThat(starter.seedList).hasSize(0);
-    assertThat(starter.simpleSnitch).isFalse();
+    starter.simpleSnitch = false;
+    starter.seedList = new ArrayList<>();
 
     starter.setStargateProperties();
 
     assertThat(System.getProperty("stargate.seed_list")).isEqualTo("127.0.0.1");
     assertThat(System.getProperty("stargate.developer_mode")).isEqualTo("true");
     assertThat(System.getProperty("stargate.snitch_classname")).isEqualTo("SimpleSnitch");
   }
+
+  @Test
+  void testSetStargatePropertiesWithIPv4ListenHost() {
+    Starter starter = setupDefaultStarter();
+    starter.listenHostStr = "127.0.0.1";
+
+    starter.setStargateProperties();
+
+    assertThat(System.getProperty("stargate.listen_address")).isEqualTo("127.0.0.1");
+  }
+
+  @Test
+  void testSetStargatePropertiesWithInvalidIPv4ListenHost() {
+    Starter starter = setupDefaultStarter();
+    starter.listenHostStr = "127.0.999.1";
+
+    RuntimeException thrown =
+        assertThrows(
+            IllegalArgumentException.class,
+            starter::setStargateProperties,
+            "Expected setStargateProperties() to throw RuntimeException");
+
+    assertThat(thrown.getMessage()).isEqualTo("--listen must be a valid IPv4 or IPv6 address");
+  }
+
+  @Test
+  void testSetStargatePropertiesWithIPv6ListenHost() {
+    Starter starter = setupDefaultStarter();
+    starter.listenHostStr = "2001:0db8:85a3:0000:0000:8a2e:0370:7334";
+
+    starter.setStargateProperties();
+
+    assertThat(System.getProperty("stargate.listen_address"))
+        .isEqualTo("2001:0db8:85a3:0000:0000:8a2e:0370:7334");
+  }
+
+  @Test
+  void testSetStargatePropertiesWithInvalidIPv6ListenHost() {
+    Starter starter = setupDefaultStarter();
+    starter.listenHostStr = "2001:0db8:85a3:x:0000:8a2e:0370:7334";
+
+    RuntimeException thrown =
+        assertThrows(
+            IllegalArgumentException.class,
+            starter::setStargateProperties,
+            "Expected setStargateProperties() to throw RuntimeException");
+
+    assertThat(thrown.getMessage()).isEqualTo("--listen must be a valid IPv4 or IPv6 address");
+  }
+
+  private Starter setupDefaultStarter() {
```

Comment:
Could be made `@BeforeEach`, I guess, to save calling it in each test method.

Reply:
True